src/JuMP.jl

```diff
@@ -346,10 +352,45 @@ optimizer is set and unsupported constraints are automatically bridged
 to equivalent supported constraints when an appropriate transformation is
 available.
 """
-bridge_constraints(model::Model) = moi_bridge_constraints(backend(model))
-# The type of backend(model) is unknown so we directly redirect to another
-# function.
+function bridge_constraints(model::Model)
+    # The type of `backend(model)` is unknown so we directly redirect to another
+    # function.
+    return moi_bridge_constraints(backend(model))
+end
 
+function moi_add_bridge(model::Nothing,
+                        BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+    # No optimizer is attached, the bridge will be added when one is attached
+end
+function moi_add_bridge(model::MOI.ModelLike,
+                        BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+    error("Cannot add bridge if `bridge_constraints` was set to `false` in the",
+          " `Model` constructor.")
+end
+function moi_add_bridge(bridge_opt::MOI.Bridges.LazyBridgeOptimizer,
+                        BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+    MOI.Bridges.add_bridge(bridge_opt, BridgeType{Float64})
+end
+function moi_add_bridge(caching_opt::MOIU.CachingOptimizer,
+                        BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+    moi_add_bridge(caching_opt.optimizer, BridgeType)
+end
+
+"""
+     add_bridge(model::Model,
+                BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+
+Add `BridgeType` to the list of bridges that can be used to transform
+unsupported constraints into an equivalent formulation using only constraints
+supported by the optimizer.
+"""
+function add_bridge(model::Model,
+                    BridgeType::Type{<:MOI.Bridges.AbstractBridge})
+    push!(model.bridge_types, BridgeType)
+    # The type of `backend(model)` is unknown so we directly redirect to another
+    # function.
```

Comment:
What about:

The type of `backend(model)` is not type-stable, so we use a function barrier (`moi_add_bridge`) to improve performance.